test/helpers/tenant/network_zone.go

```diff
@@ -0,0 +1,108 @@
+//go:build e2e
+
+package tenant
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+type FallbackMode string
+
+const (
+	FallbackAnyActiveGate FallbackMode = "ANY_ACTIVE_GATE"
+	FallbackNone          FallbackMode = "NONE"
+	FallbackDefaultZone   FallbackMode = "ONLY_DEFAULT_ZONE"
+)
+
+type networkZoneRequestBody struct {
+	AlternativeZones []string `json:"alternativeZones,omitempty"`
+	FallbackMode     string   `json:"fallbackMode,omitempty"`
+	Id               string   `json:"id,omitempty"`
+}
+
+func CreateNetworkZone(secret Secret, networkZone string, alternativeZones []string, fallbackMode FallbackMode) features.Func {
+	return func(ctx context.Context, t *testing.T, envConfig *envconf.Config) context.Context {
+		// API documentation
+		// https://www.dynatrace.com/support/help/dynatrace-api/environment-api/network-zones/put-network-zone
+
+		nzApiUrl := fmt.Sprintf("%s/v2/networkZones/%s", secret.ApiUrl, networkZone)
+
+		request := networkZoneRequestBody{
+			AlternativeZones: alternativeZones,
+			FallbackMode:     string(fallbackMode),
+		}
+
+		body, err := json.Marshal(request)
+		require.NoError(t, err)
+
+		client := &http.Client{}
+		req, err := http.NewRequest(http.MethodPut, nzApiUrl, bytes.NewReader(body))
+		require.NoError(t, err)
+
+		req.Header.Add("Authorization", "Api-Token "+secret.ApiToken)
+		req.Header.Add("Content-Type", "application/json")
+
+		resp, err := client.Do(req)
+		defer func() { _ = resp.Body.Close() }()
```

Comment:
`Do` may return an error (`nil` response) so we should swap lines 58 and 59 in place or check if `resp` is not `nil` in the `defer` statement